core/network/src/message.rs

```diff
@@ -216,6 +225,8 @@ pub mod generic {
 	pub struct Status<Hash, Number> {
 		/// Protocol version.
 		pub version: u32,
+		/// Minimum supported version.
+		pub min_supported_version: u32,
```

Comment:
As discussed in the chat: this part only breaks compatibility between this code base and (non-authority) full-nodes in the existing network. Maybe we can isolate this change to make it easier to migrate the other changes onto live networks?

Reply:
Changes to the gossip message are breaking as well. Maintaining backwards compatibility is a pain currently because peers would just disconnect if other party has higher protocol version. This exact change is there to address that.